app/controllers/access_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Define actions for handling user access.
+class AccessController < ApplicationController
+  include AccessHelper
+
+  before_action :confirm_logged_in, only: %i[menu]
+
+  # GET /access/menu
+  # Show admin menu panel.
+  def menu; end
+
+  # GET /access/login
+  # Show login form.
+  def login; end
+
+  # POST /access/login
+  # Attempt to login user or show error message.
+  def attempt_login
+    if params[:username].present? && params[:password].present?
+      user = Admin.where(username: params[:username]).first
+      if user
+        authorized_user = user.authenticate(params[:password])
+      end
+    end
+
+    if authorized_user
```

Comment:
you could remove this autorized_user variable and use the user.authenticate(params[:password])